userspace/libsinsp/filter.cpp

```diff
@@ -415,6 +415,8 @@ bool flt_compare(cmpop op, ppm_param_type type, void* operand1, void* operand2,
 		return flt_compare_buffer(op, (char*)operand1, (char*)operand2, op1_len, op2_len);
 	case PT_DOUBLE:
 		return flt_compare_double(op, *(double*)operand1, *(double*)operand2);
+	case PT_CHARBUFARRAY:
+		throw sinsp_exception("charbuf-array type can't be compared to single values");
```

Comment:
How about a comment stating that the caller ensures that flt_compare is never called with a PT_CHARBUFARRAY? I originally thought that a badly written filter might result in this exception and thought it was too aggressive.